thunder/distributed/prims.py

```diff
@@ -184,14 +184,7 @@ def synchronize_meta(
             # Assuming that the sharding is done on the first dimension
             # See [FSDP Sharding] in distributed/__init__.py
             unsharded_shape = a.shape[0] * group.size(), *a.shape[1:]
-            # NOTE: We explicitly pass the `a.requires_grad` as we want to propogate the requires grad correctly on `nn.Parameter`'s
-            # as it maybe queried later.
-            return TensorProxy(
-                shape=unsharded_shape,
-                like=a,
-                distparallel_type=DistParallelType.REPLICATED,
-                requires_grad=a.requires_grad,
-            )
```

Comment:
Should this change be preserved?

Reply:
`requires_grad` should be propagated only in the meta function of the grad transform and tested. It's incorrect to rely on this attribute for intermediates because there's no guarantee it's used consistently. 

I agree with you here that there's no need to revert this change. We only need to revert the changes that caused performance regression.